Foxtrot/Driver/Rewriting/Utils/TypeNodeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class TypeNodeExtensions
+    {
+        public static int TemplateArgumentsCount(this TypeNode typeNode)
+        {
+            Contract.Requires(typeNode != null);
+
+            return typeNode.TemplateArguments == null ? 0 : typeNode.TemplateArguments.Count;
+        }
+
+        public static int CountOrDefault(this TypeNodeList list)
```

Comment:
I now that this solution is far from perfect AND it would be better to use new language feature one day. But I disagree that this solution brings no value.

Compare two different versions:

```c#
                    var templateArgCount = (originalInstanceMethod.TemplateArguments == null)
                        ? 0
                        : originalInstanceMethod.TemplateArguments.Count;

                    for (int i = 0; i < templateArgCount; i++)
                    {
                        args.Add(originalInstanceMethod.TemplateArguments[i]);
                    }
```

and

```c#
                    for (int i = 0; i < originalInstanceMethod.TemplateArguments.CountOrDefault(); i++)
                    {
                        args.Add(originalInstanceMethod.TemplateArguments[i]);
                    }
```

And please note that there is a few places like that in the code. 

I'm OK to add TODO to change this after migration to C# 6.0, but *I'm not OK* to remove to old version.